[2025-10-05T03:20:30.162+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-10-05T03:20:30.173+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: forest_cover_mlops_pipeline.train_and_log_models scheduled__2025-10-05T03:15:00+00:00 [queued]>
[2025-10-05T03:20:30.179+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: forest_cover_mlops_pipeline.train_and_log_models scheduled__2025-10-05T03:15:00+00:00 [queued]>
[2025-10-05T03:20:30.180+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 3
[2025-10-05T03:20:30.188+0000] {taskinstance.py:2888} INFO - Executing <Task(_PythonDecoratedOperator): train_and_log_models> on 2025-10-05 03:15:00+00:00
[2025-10-05T03:20:30.196+0000] {standard_task_runner.py:72} INFO - Started process 1714 to run task
[2025-10-05T03:20:30.200+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'forest_cover_mlops_pipeline', 'train_and_log_models', 'scheduled__2025-10-05T03:15:00+00:00', '--job-id', '26', '--raw', '--subdir', 'DAGS_FOLDER/forest_cover_dag.py', '--cfg-path', '/home/airflow/tmp/tmpchruw2m5']
[2025-10-05T03:20:30.201+0000] {standard_task_runner.py:105} INFO - Job 26: Subtask train_and_log_models
[2025-10-05T03:20:30.264+0000] {task_command.py:467} INFO - Running <TaskInstance: forest_cover_mlops_pipeline.train_and_log_models scheduled__2025-10-05T03:15:00+00:00 [running]> on host 251551b582e0
[2025-10-05T03:20:30.544+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='mlops-team-5' AIRFLOW_CTX_DAG_ID='forest_cover_mlops_pipeline' AIRFLOW_CTX_TASK_ID='train_and_log_models' AIRFLOW_CTX_EXECUTION_DATE='2025-10-05T03:15:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-10-05T03:15:00+00:00'
[2025-10-05T03:20:30.545+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-10-05T03:20:30.698+0000] {logging_mixin.py:190} INFO - Entrenando 3 modelos con GridSearchCV...
[2025-10-05T03:20:30.699+0000] {logging_mixin.py:190} INFO - 
==================================================
[2025-10-05T03:20:30.699+0000] {logging_mixin.py:190} INFO - Modelo: RandomForest
[2025-10-05T03:20:30.699+0000] {logging_mixin.py:190} INFO - ==================================================
[2025-10-05T03:20:30.715+0000] {logging_mixin.py:190} WARNING - 2025/10/05 03:20:30 WARNING mlflow.utils.git_utils: Failed to import Git (the Git executable is probably not on your PATH), so Git SHA is not available. Error: Failed to initialize: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh(<full-path-to-git-executable>)

All git commands will error until this is rectified.

This initial message can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|silent|none|n|0: for no message or exception
    - warn|w|warning|log|l|1: for a warning message (logging level CRITICAL, displayed by default)
    - error|e|exception|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet
[2025-10-05T03:20:30.780+0000] {warnings.py:110} WARNING - /home/airflow/.local/lib/python3.11/site-packages/joblib/parallel.py:1379: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = self._backend.configure(

[2025-10-05T03:20:30.780+0000] {logging_mixin.py:190} INFO - Fitting 3 folds for each of 12 candidates, totalling 36 fits
[2025-10-05T03:20:58.533+0000] {logging_mixin.py:190} INFO -   Mejores par√°metros: {'classifier__max_depth': None, 'classifier__min_samples_split': 2, 'classifier__n_estimators': 100}
[2025-10-05T03:21:01.782+0000] {logging_mixin.py:190} INFO -   Train Accuracy: 1.0000
[2025-10-05T03:21:01.782+0000] {logging_mixin.py:190} INFO -   Test Accuracy: 0.9522
[2025-10-05T03:21:01.782+0000] {logging_mixin.py:190} INFO -   Test F1 (weighted): 0.9519
[2025-10-05T03:21:01.782+0000] {logging_mixin.py:190} INFO -   CV Score: 0.9456 (+/- 0.0026)
[2025-10-05T03:21:03.633+0000] {logging_mixin.py:190} WARNING - 2025/10/05 03:21:03 WARNING mlflow.models.model: Model logged without a signature and input example. Please set `input_example` parameter when logging the model to auto infer the model signature.
[2025-10-05T03:21:03.687+0000] {credentials.py:1147} INFO - Found credentials in environment variables.
[2025-10-05T03:21:03.910+0000] {logging_mixin.py:190} WARNING - Registered model 'forest_cover_randomforest' already exists. Creating a new version of this model...
[2025-10-05T03:21:03.928+0000] {logging_mixin.py:190} WARNING - 2025/10/05 03:21:03 INFO mlflow.store.model_registry.abstract_store: Waiting up to 300 seconds for model version to finish creation. Model name: forest_cover_randomforest, version 3
[2025-10-05T03:21:03.928+0000] {logging_mixin.py:190} WARNING - Created version '3' of model 'forest_cover_randomforest'.
[2025-10-05T03:21:03.937+0000] {logging_mixin.py:190} WARNING - 2025/10/05 03:21:03 INFO mlflow.tracking._tracking_service.client: üèÉ View run RandomForest_batch_4 at: http://mlflow:5000/#/experiments/1/runs/8ced60e7bc8b4fc78a59045b3039fbce.
[2025-10-05T03:21:03.938+0000] {logging_mixin.py:190} WARNING - 2025/10/05 03:21:03 INFO mlflow.tracking._tracking_service.client: üß™ View experiment at: http://mlflow:5000/#/experiments/1.
[2025-10-05T03:21:03.948+0000] {logging_mixin.py:190} INFO - 
==================================================
[2025-10-05T03:21:03.948+0000] {logging_mixin.py:190} INFO - Modelo: GradientBoosting
[2025-10-05T03:21:03.948+0000] {logging_mixin.py:190} INFO - ==================================================
[2025-10-05T03:21:04.052+0000] {warnings.py:110} WARNING - /home/airflow/.local/lib/python3.11/site-packages/joblib/parallel.py:1379: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = self._backend.configure(

[2025-10-05T03:21:04.053+0000] {logging_mixin.py:190} INFO - Fitting 3 folds for each of 8 candidates, totalling 24 fits
[2025-10-05T03:24:59.023+0000] {logging_mixin.py:190} INFO -   Mejores par√°metros: {'classifier__learning_rate': 0.1, 'classifier__max_depth': 5, 'classifier__n_estimators': 100}
[2025-10-05T03:25:47.870+0000] {logging_mixin.py:190} INFO -   Train Accuracy: 0.9796
[2025-10-05T03:25:47.871+0000] {logging_mixin.py:190} INFO -   Test Accuracy: 0.9333
[2025-10-05T03:25:47.871+0000] {logging_mixin.py:190} INFO -   Test F1 (weighted): 0.9328
[2025-10-05T03:25:47.871+0000] {logging_mixin.py:190} INFO -   CV Score: 0.9334 (+/- 0.0028)
[2025-10-05T03:25:49.327+0000] {logging_mixin.py:190} WARNING - 2025/10/05 03:25:49 WARNING mlflow.models.model: Model logged without a signature and input example. Please set `input_example` parameter when logging the model to auto infer the model signature.
[2025-10-05T03:25:49.398+0000] {logging_mixin.py:190} WARNING - Registered model 'forest_cover_gradientboosting' already exists. Creating a new version of this model...
[2025-10-05T03:25:49.416+0000] {logging_mixin.py:190} WARNING - 2025/10/05 03:25:49 INFO mlflow.store.model_registry.abstract_store: Waiting up to 300 seconds for model version to finish creation. Model name: forest_cover_gradientboosting, version 2
[2025-10-05T03:25:49.417+0000] {logging_mixin.py:190} WARNING - Created version '2' of model 'forest_cover_gradientboosting'.
[2025-10-05T03:25:49.424+0000] {logging_mixin.py:190} WARNING - 2025/10/05 03:25:49 INFO mlflow.tracking._tracking_service.client: üèÉ View run GradientBoosting_batch_4 at: http://mlflow:5000/#/experiments/1/runs/d4256272ac2b45d8b498adf554668417.
[2025-10-05T03:25:49.425+0000] {logging_mixin.py:190} WARNING - 2025/10/05 03:25:49 INFO mlflow.tracking._tracking_service.client: üß™ View experiment at: http://mlflow:5000/#/experiments/1.
[2025-10-05T03:25:49.434+0000] {logging_mixin.py:190} INFO - 
==================================================
[2025-10-05T03:25:49.434+0000] {logging_mixin.py:190} INFO - Modelo: LogisticRegression
[2025-10-05T03:25:49.435+0000] {logging_mixin.py:190} INFO - ==================================================
[2025-10-05T03:25:49.486+0000] {warnings.py:110} WARNING - /home/airflow/.local/lib/python3.11/site-packages/joblib/parallel.py:1379: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = self._backend.configure(

[2025-10-05T03:25:49.487+0000] {logging_mixin.py:190} INFO - Fitting 3 folds for each of 6 candidates, totalling 18 fits
[2025-10-05T03:26:13.942+0000] {warnings.py:110} WARNING - /home/airflow/.local/lib/python3.11/site-packages/sklearn/linear_model/_sag.py:349: ConvergenceWarning: The max_iter was reached which means the coef_ did not converge
  warnings.warn(

[2025-10-05T03:26:16.990+0000] {warnings.py:110} WARNING - /home/airflow/.local/lib/python3.11/site-packages/sklearn/linear_model/_sag.py:349: ConvergenceWarning: The max_iter was reached which means the coef_ did not converge
  warnings.warn(

[2025-10-05T03:26:20.043+0000] {warnings.py:110} WARNING - /home/airflow/.local/lib/python3.11/site-packages/sklearn/linear_model/_sag.py:349: ConvergenceWarning: The max_iter was reached which means the coef_ did not converge
  warnings.warn(

[2025-10-05T03:26:23.191+0000] {logging_mixin.py:190} INFO -   Mejores par√°metros: {'classifier__C': 10.0, 'classifier__solver': 'lbfgs'}
[2025-10-05T03:26:32.114+0000] {logging_mixin.py:190} INFO -   Train Accuracy: 0.7650
[2025-10-05T03:26:32.115+0000] {logging_mixin.py:190} INFO -   Test Accuracy: 0.7601
[2025-10-05T03:26:32.117+0000] {logging_mixin.py:190} INFO -   Test F1 (weighted): 0.7502
[2025-10-05T03:26:32.118+0000] {logging_mixin.py:190} INFO -   CV Score: 0.7643 (+/- 0.0040)
[2025-10-05T03:26:33.804+0000] {logging_mixin.py:190} WARNING - 2025/10/05 03:26:33 WARNING mlflow.models.model: Model logged without a signature and input example. Please set `input_example` parameter when logging the model to auto infer the model signature.
[2025-10-05T03:26:33.849+0000] {logging_mixin.py:190} WARNING - Registered model 'forest_cover_logisticregression' already exists. Creating a new version of this model...
[2025-10-05T03:26:33.863+0000] {logging_mixin.py:190} WARNING - 2025/10/05 03:26:33 INFO mlflow.store.model_registry.abstract_store: Waiting up to 300 seconds for model version to finish creation. Model name: forest_cover_logisticregression, version 2
[2025-10-05T03:26:33.863+0000] {logging_mixin.py:190} WARNING - Created version '2' of model 'forest_cover_logisticregression'.
[2025-10-05T03:26:33.869+0000] {logging_mixin.py:190} WARNING - 2025/10/05 03:26:33 INFO mlflow.tracking._tracking_service.client: üèÉ View run LogisticRegression_batch_4 at: http://mlflow:5000/#/experiments/1/runs/fdcc237d583b47bb867596790c7eb740.
[2025-10-05T03:26:33.869+0000] {logging_mixin.py:190} WARNING - 2025/10/05 03:26:33 INFO mlflow.tracking._tracking_service.client: üß™ View experiment at: http://mlflow:5000/#/experiments/1.
[2025-10-05T03:26:33.878+0000] {logging_mixin.py:190} INFO - 
==================================================
[2025-10-05T03:26:33.878+0000] {logging_mixin.py:190} INFO - ‚úì Entrenamiento completado
[2025-10-05T03:26:33.878+0000] {logging_mixin.py:190} INFO -   Mejor modelo: RandomForest (Accuracy: 0.9522)
[2025-10-05T03:26:33.878+0000] {logging_mixin.py:190} INFO - ==================================================
[2025-10-05T03:26:33.878+0000] {python.py:240} INFO - Done. Returned value was: {'results': {'RandomForest': {'test_accuracy': 0.9522375215146299, 'test_f1_weighted': 0.9518738995728053, 'cv_mean_score': 0.9456219634183364, 'best_params': {'classifier__max_depth': None, 'classifier__min_samples_split': 2, 'classifier__n_estimators': 100}}, 'GradientBoosting': {'test_accuracy': 0.9333046471600689, 'test_f1_weighted': 0.9328300616084608, 'cv_mean_score': 0.9334121273536732, 'best_params': {'classifier__learning_rate': 0.1, 'classifier__max_depth': 5, 'classifier__n_estimators': 100}}, 'LogisticRegression': {'test_accuracy': 0.7601118760757315, 'test_f1_weighted': 0.7501504556174429, 'cv_mean_score': 0.764307187052259, 'best_params': {'classifier__C': 10.0, 'classifier__solver': 'lbfgs'}}}, 'best_model': 'RandomForest', 'best_score': 0.9522375215146299, 'experiment_name': 'forest_cover_classification', 'mlflow_uri': 'http://mlflow:5000'}
[2025-10-05T03:26:33.902+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-10-05T03:26:33.902+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=forest_cover_mlops_pipeline, task_id=train_and_log_models, run_id=scheduled__2025-10-05T03:15:00+00:00, execution_date=20251005T031500, start_date=20251005T032030, end_date=20251005T032633
[2025-10-05T03:26:33.932+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-10-05T03:26:33.945+0000] {taskinstance.py:3900} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2025-10-05T03:26:33.947+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
